src/rdb_protocol/datum_stream.cc

```diff
@@ -2014,7 +2016,8 @@ range_datum_stream_t::next_raw_batch(env_t *, const batchspec_t &batchspec) {
                             batchspec.with_at_most(500).to_batcher();
 
     while (!is_exhausted()) {
-        double next = safe_to_double(start++);
+        double next = safe_to_double(start);
+        start+=step;
```

Comment:
Per our usual style, there should be spaces around the operator: `start += step;`